react/PromotionBanner.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC } from 'react'
+import PropTypes from 'prop-types'
+import { injectIntl, InjectedIntlProps } from 'react-intl'
+
+import { Button } from 'vtex.styleguide'
+import { usePWA } from 'vtex.store-resources/PWAContext'
+
+import setWebAppData from './utils/webAppIndexedDB'
+
+const propTypes = {
+  imageUrl: PropTypes.string,
+  type: PropTypes.oneOf(['install', 'push-notification']),
+}
+
+const CONSTANTS = {
+  INSTALL_TITLE: 'store/promotion-banner.install.title',
+  INSTALL_DESCRIPTION: 'store/promotion-banner.install.description',
+  INSTALL_ACCEPT_BUTTON_LABEL: 'store/promotion-banner.install.button-accept.label',
+  PUSH_NOTIFICATION_TITLE: 'store/promotion-banner.push-notification.title',
+  PUSH_NOTIFICATION_DESCRIPTION: 'store/promotion-banner.push-notification.description',
+  PUSH_NOTIFICATIONS_ACCEPT_BUTTON_LABEL: 'store/promotion-banner.push-notification.button-accept.label',
+  DISMISS_BUTTON_LABEL: 'store/promotion-banner.button-dismiss.label'
+}
+
+type Props = PropTypes.InferProps<typeof propTypes>
+
+const PromotionBanner: FC<Props & InjectedIntlProps> = ({ intl, type = 'install' }) => {
+  const { showInstallPrompt = null } = usePWA() || {}
+
+  const handleDismiss = async () => {
+    if(type === 'install')
+      setWebAppData('appInstallDismissed')
```

Comment:
It might makes sense to put all these string in the `CONSTANTS`

Reply:
Done!